trie-db/src/triedbmut.rs

```diff
@@ -2082,15 +2325,33 @@ where
 		Ok(old_val)
 	}
 
-	fn remove(&mut self, key: &[u8]) -> Result<Option<Value<L>>, TrieHash<L>, CError<L>> {
+	pub fn remove(&mut self, key: &[u8]) -> Result<Option<Value<L>>, TrieHash<L>, CError<L>> {
+		self.remove_with_child_changes(key, None)
+	}
+
+	pub fn remove_with_child_changes(
```

Comment:
change from arkpar/new-trie

Reply:
`remove_with_tree_ref`

I guess eventuall we won't need this? The point of tree references is that the database handles dereferencing them automatically.